### PR TITLE
Great Rework Part 15 -- Code Blocks, Tabs, Titles, and Interaction

### PR DESCRIPTION
**Pull request by TheOnlyRew**
This commit add JS generation and interaction for everything Code Block. Titles will be added to fences that are given a language. Runs of such blocks will be compressed into a single Tabbed Code Block.

I must say that I am... not happy with the way this change set worked out. Names are too verbose even for my tastes, and there is way more CoffeeScript generating and interacting with the Code Blocks than I would have liked. Criticism would be very welcome here.

~~_Note that at the time of writing, I have yet to lick the canvas. I'll be coming back in and cleaning up comment styles, typos, etc.. I don't think I'll be able to make the kind of structural changes required to actually clean this code up as I can no longer see the forest for the trees._~~
Edit: Comments have been cleaned up pretty substantially. Still not 100% about the structure, but... I'm feeling better about it.

---

**THIS PR IS A WORK IN PROGRESS**
I've opened this as part of a set of stacked PRs so that I can more cleanly and more quickly share the work that I'm doing. Any changes made to this branch will likely come in the form of a force push, so local updates should be performed with

```
git checkout great_rework/15/code_blocks_tabs_titles
git fetch origin great_rework/15/code_blocks_tabs_titles
git reset origin/great_rework/15/code_blocks_tabs_titles --hard
```

Please feel free to comment on anything you see here; offer suggestions, provide corrections, raise concerns. the whole point of this exercise is to engender communication.

But _**do not merge this PR**_.

### CODE REVIEW

File: dynamic/js/basho/code-blocks.coffee [hunk 1]
```diff
@@ -0,0 +1,462 @@
...
+        . . .
+      </code></pre>
+    </div>
+  </div>
+###
+
+#TODO: strict mode enables a pretty large number of runtime checks. We probably
+#      want to turn it off when we deploy to prod.
+#      Perhaps we can figure out some way to enable it when building debug?
+'use strict'
```
Comment (lukebakken) [inline dynamic/js/basho/code-blocks.coffee]: FWIW, strict mode is always on in the Riak node.js client. I doubt there would be any noticable performance change in the browser, too.

**Comment by lukebakken** [comment]
Any way to see this in action?

**Comment by TheOnlyRew** [comment]
@lukebakken checkout the great_rework/WIP branch, `rake build:debug; rake hugo:server`, and open up localhost:1313/sandbox/2.1.4/reference_page
That does remind me that I need to redo some of the older commits in this historical line to get the changes working for the main site. Thought I did that already, but obviously something's not working.

**Comment by lukebakken** [comment]
OK, I pulled it up. Looks good to me.

**Comment by TheOnlyRew** [comment]
After looking at how much I munged the history of this change set, and after looking at how tightly coupled and _big_ this change set is, I'm throwing in the towel with this experiment.

I think it may have worked if there was iterative review to go along with the iterative progress but I failed to secure that kind of review, and I let the stacked PRs fall out of date with the head of development. At this point, there's too much entropy to make cleaning the history a worthwhile task.

Moving forward I'm going to consider building a "false history" that will (hopefully) accomplish the goals of this stacked PR set; smaller, logically grouped change sets that make reviews possible.

**Pull request closed without merging**